src/components/VmDialog/index.js

```diff
@@ -808,7 +809,7 @@ VmDialog.propTypes = {
 
 export default connect(
   (state) => ({
-    clusters: state.clusters,
+    clusters: getFilteredClusters(state.clusters),
```

Comment:
No, in redux store clusters must be unfiltered, because edit != view. For example UserRole, or ReadOnlyAdmin (this role can only see everything but not edit).

Reply:
Ok so the full cluster list is still needed because a user can still see the cluster.  Cool.

How about adding an item to the internal Cluster type to hold the results of the filter check?  Fetch the clusters (with permissions) and then in the transform, check the permissions and flag each cluster as "view" or "edit" (or something similar).  Filtering would still need to be done in the edit places, but not the permission checks.

You could also do that for VMs ... boolean field 'canUserEdit'